controllers/state_of_the_world.go

```diff
@@ -234,3 +371,18 @@ func (e *EventLogger) Log(ctx context.Context, resourceEvents []controller.Resou
 
 	return nil
 }
+
+func isObjectOwnedByKind(o client.Object, ownerGroup, ownerKind string) bool {
```

Comment:
Could receive a `schema.GroupKind` instead of 2 strings, and thus help with consistently defining and reusing all those XGroupKind variables, without any weird stuff.

Reply:
That's a good idea, I'll update 👍